internal/subscriber/subscriber.go

```diff
@@ -27,8 +27,8 @@ type ISubscriber interface {
 	GetSubscription() string
 	// not exporting acknowledge() and  modifyAckDeadline() intentionally so that
 	// all operations happen over the channel
-	acknowledge(ctx context.Context, req *AckMessage)
-	modifyAckDeadline(ctx context.Context, req *ModAckMessage)
```

Comment:
?

Reply:
Since these methods are not exported, there was some issue with mock generated. generated mock wasn't assignable to the interface. After commenting this it worked. mostly a issue in golang

Reply:
https://github.com/razorpay/metro/pull/201/files#diff-1d4ef1c48c1b1aff0ea88f681a7e7c463724dd56ce27f6bbd2d17f804c11a809R188

This fails to handle the type with unexported method defined in the interface